src/lib.rs

```diff
@@ -335,16 +336,17 @@ impl Bump {
         old_size_with_footer: Option<usize>,
         requested_layout: Option<Layout>,
         prev: Option<NonNull<ChunkFooter>>,
-    ) -> NonNull<ChunkFooter> {
+    ) -> Option<NonNull<ChunkFooter>> {
         unsafe {
             // As a sane default, we want our new allocation to be about twice as
             // big as the previous allocation
             let mut new_size_without_footer =
                 if let Some(old_size_with_footer) = old_size_with_footer {
                     let old_size_without_footer = old_size_with_footer - FOOTER_SIZE;
-                    old_size_without_footer
-                        .checked_mul(2)
-                        .unwrap_or_else(|| oom())
+                    match old_size_without_footer.checked_mul(2) {
+                        Some(size) => size,
+                        None => return None,
```

Comment:
I think this whole match can become `old_size_without_footer.checked_mul(2)?` and use `?` propagation.